src/integrations/feed/configs.ts

```diff
@@ -129,31 +129,43 @@ type FeedLofnConfigGeneratorOptions = {
 export class FeedLofnConfigGenerator implements FeedConfigGenerator {
   private readonly lofnClient: ILofnClient;
   private readonly opts: FeedLofnConfigGeneratorOptions;
+  private readonly feedPreferencesConfigGenerator: FeedPreferencesConfigGenerator;
 
   constructor(lofnClient: ILofnClient, opts: FeedLofnConfigGeneratorOptions) {
     this.lofnClient = lofnClient;
     this.opts = opts;
+    this.feedPreferencesConfigGenerator = new FeedPreferencesConfigGenerator(
+      {},
+      opts,
+    );
   }
 
   async generate(
     ctx: Context,
     opts: DynamicConfig,
   ): Promise<FeedConfigGeneratorResult> {
     return runInSpan('FeedLofnConfigGenerator', async () => {
-      const lofnConfig = await this.lofnClient.fetchConfig({
-        user_id: opts.user_id,
-        feed_version: this.opts.feed_version,
-      });
-
-      const feedPreferencesConfigGenerator = new FeedPreferencesConfigGenerator(
+      const [lofnConfig, preferencesConfig] = await Promise.all([
+        this.lofnClient.fetchConfig({
+          user_id: opts.user_id,
+          feed_version: this.opts.feed_version,
+        }),
+        this.feedPreferencesConfigGenerator.generate(ctx, opts),
+      ]);
+
+      const config = getDefaultConfig(
+        {
+          ...preferencesConfig.config,
+          total_pages: lofnConfig.config.total_pages,
```

Comment:
We don't really need to take lofn total pages. We can keep it owned by API just like all other pagination params

Reply:
`page_size` as well?

Reply:
These should take from API?
- fresh_page_size
- page_size
- total_pages

Reply:
yes